lib/vintage_net/connectivity/internet_checker.ex

```diff
@@ -165,6 +167,15 @@ defmodule VintageNet.Connectivity.InternetChecker do
     %{state | check_logic: CheckLogic.check_failed(state.check_logic)}
   end
 
+  defp pet_pm_watchdog(%{status: :internet, ifname: ifname} = state) do
+    :ok = PMControl.pet_watchdog(ifname)
+    state
+  end
+
+  defp pet_pm_watchdog(%{status: _} = state) do
+    state
+  end
+
```

Comment:
```suggestion
  defp pet_pm_watchdog(%{status: status, ifname: ifname} = state) when status in [:lan, :internet] do
    PMControl.pet_watchdog(ifname)
    state
  end

  defp pet_pm_watchdog(%{status: _} = state) do
    state
  end
```

I switched the logic to pet the power manager watchdog when in LAN mode as well, since that could be an ok mode for the WiFi module (or whatever) to be in and we wouldn't want to reboot it.